### PR TITLE
chore(repository-json-schema): add JSON schema type tests

### LINKED ISSUE #971: Add in tests for JSON Schemas being produced
- JSON Schemas produced by `repository-json-schema` should be validated against this meta-schema (used to validate the schema itself): http://json-schema.org/schema. More information here: http://json-schema.org/specification.html#meta-schemas.

## Acceptance Criteria
- [ ] write tests for _types_ of JSON Schema in `repository-json-schema`
- [ ] validate JSON Schemas produced in the tests of `repository-json-schema` with a JSON-Schema-validating-JSON Schema

### PR DESCRIPTION
**Pull request by shimks**
- closes #971 

## Checklist

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/repository-json-schema/test/integration/build-schema.integration.ts [hunk 1]
```diff
@@ -56,8 +60,12 @@ describe('build-schema', () => {
           bar: number;
         }
 
-        expect(modelToJsonSchema(Empty)).to.eql({});
-        expect(modelToJsonSchema(NoModelMeta)).to.eql({});
+        const emptyJson = modelToJsonSchema(Empty);
+        const noModelMetaJson = modelToJsonSchema(NoModelMeta);
+        expect(emptyJson).to.eql({});
```
Comment (virkt25) [inline packages/repository-json-schema/test/integration/build-schema.integration.ts]: This seems redundant. Won't `expectValidJsonSchema` check if it's `{}` or not?

File: packages/repository-json-schema/test/unit/json-schema.unit.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import {JsonSchema} from '../../src';
```
Comment (virkt25) [inline packages/repository-json-schema/test/unit/json-schema.unit.ts]: Move import below the Copyright notice.

File: packages/repository-json-schema/test/integration/build-schema.integration.ts (hunk 1 above)
Comment (shimks) [reply packages/repository-json-schema/test/integration/build-schema.integration.ts]: I guess it is redundant. On the other hand though, what if someday they decide `{}` is not a valid JSON Schema? In that case, we'd want this to test to fail while still maintaining that `emptyJson` is an empty object and that `{}` is not a valid JSON Schema

Comment (virkt25) [reply packages/repository-json-schema/test/integration/build-schema.integration.ts]: Unlikely, but I'm ok with the redundancy.

File: packages/repository-json-schema/test/unit/json-schema.unit.ts (hunk 2 above)
Comment (shimks) [reply packages/repository-json-schema/test/unit/json-schema.unit.ts]: done

File: packages/repository-json-schema/test/integration/build-schema.integration.ts [hunk 3]
```diff
@@ -335,6 +355,18 @@ describe('build-schema', () => {
         }).to.throw(/type is defined as an array/);
       });
     });
+
+    function expectValidJsonSchema(schema: JsonSchema) {
```
[the remark this answers is not in the document]
Comment (shimks) [reply packages/repository-json-schema/test/integration/build-schema.integration.ts]: :thinking: This means that we'd have to have `@loopback/testlab` and `ajv` as dependencies for `@loopback/repository`. Looking at https://bundlephobia.com/result?p=ajv@6.5.0, would you consider `ajv` to be sizable?

[the remark this answers is not in the document]
Comment (virkt25) [reply packages/repository-json-schema/test/integration/build-schema.integration.ts]: It's not that big imo but I'm ok not exporting this as a utility function.

[the remark this answers is not in the document]
Comment (bajtos) [reply packages/repository-json-schema/test/integration/build-schema.integration.ts]: @shimks I think @virkt25 was suggesting to export `expectValidJsonSchema` from `repository-json-schema`, not `testlab` package?

At one hand, a readily-available helper for verifying JSON Schema may be helpful outside of this module tests too. On the other hand, exporting a test helper from a non-test-helper module is tricky as we may need to move dev-dependencies to regular dependencies. As for `testlab`, it should not become a kitchen-sink for everything testing related.

In the light of what I wrote above, I think we should not be exporting this utility function.

**Pull request merged**